.github/workflows/publish-to-npm.yml

```diff
@@ -0,0 +1,23 @@
+name: npm-publish
+on:
+  push:
+    branches:
+      - main 
+jobs:
+  npm-publish:
+    name: npm-publish
+    if: github.repository == 'adobe/plugin-on-fetch'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@main
+    - name: Set up Node.js
+      uses: actions/setup-node@main
+      with:
+        node-version: 18
+    - name: Publish if version has been updated
+      uses: mkiki/npm-publish-action@c4315ef5790b7bcec2cbb75b34e37681a409d78d
+    env: 
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 
+        NPM_AUTH_TOKEN: ${{ secrets.ADOBE_BOT_NPM_TOKEN }} 
+        NPM_AUTH: ${{ secrets.ADOBE_BOT_NPM_TOKEN }} 

```

Comment:
@AjazSumaiya, I am not entirely if we need to set the access to publish in the npm-publish just like we have in [here](https://github.com/adobe/aio-cli-plugin-api-mesh/blob/main/.github/workflows/publish-to-npm.yml#L20) although we have the publishConfig set to "public" in [package.json](https://github.com/adobe/plugin-on-fetch/pull/1/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L5)
@revanth0212

Reply:
Hi @revanth0212 , yes as per my understanding setting publishConfig to "public" should have been sufficient, but since our package is scoped one, by default which is private, we'll need to set the access level in publish-npm.yml file as well because npm requires first publish of the package to be explicitly set as _"public"_ for packages that are default _"private"._ 

> Scoped packages are not public by default. You will need to specify --access public with the initial npm publish command. This will publish the package and set access to public as if you had run npm access public after publishing. You do not need to do this when publishing new versions of an existing scoped package. https://docs.npmjs.com/cli/v9/using-npm/scope

so, for subsequent publishes package.json publishConfig is sufficient but for first publish action we need it to be set here as well when github action runs npm publish command.